workflow/metrics/leader.go

```diff
@@ -0,0 +1,44 @@
+package metrics
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/metric"
+)
+
+type LeaderStateCallback func() bool
+
+type leaderGauge struct {
+	callback LeaderStateCallback
+	gauge    *instrument
+}
+
+func addIsLeader(ctx context.Context, m *Metrics) error {
+	const nameLeader = `leader`
+	err := m.createInstrument(int64ObservableGauge,
+		nameLeader,
+		"Emits 1 if this is the leader when leader elections are enabled, or 0 otherwise. Always 1 when leader elections are disabled.",
+		"{leader}",
+		withAsBuiltIn(),
+	)
+	if err != nil {
+		return err
+	}
+	if m.callbacks.LeaderState != nil {
+		lGauge := leaderGauge{
+			callback: m.callbacks.LeaderState,
+			gauge:    m.allInstruments[nameLeader],
+		}
+		return m.allInstruments[nameLeader].registerCallback(m, lGauge.update)
+	}
+	return nil
```

Comment:
would probably make sense to invert this conditional (and all other equivalent instances of it), no? Since it would only be `nil` for the test cases, right?

```suggestion
	if m.callbacks.LeaderState == nil {
		return nil
	}    

	lGauge := leaderGauge{
		callback: m.callbacks.LeaderState,
		gauge:    m.allInstruments[nameLeader],
	}
	return m.allInstruments[nameLeader].registerCallback(m, lGauge.update)
```